boxes/generators/dicebox.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+# Copyright (C) 2022 Erik Snider (SniderThanYou@gmail.com)
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from boxes import *
+
+
+class DiceBox(Boxes):
+    """Box with lid and integraded hinge for storing dice"""
+
+    ui_group = "Misc"
+
+    def __init__(self):
+        Boxes.__init__(self)
+        self.addSettingsArgs(
+            edges.FingerJointSettings,
+            surroundingspaces=2.0)
+        self.addSettingsArgs(
+            edges.ChestHingeSettings,
+            finger_joints_on_box=True,
+            finger_joints_on_lid=True)
+        self.buildArgParser(
+            x=100,
+            y=100,
+            h=18,
+            outside=True)
+        self.argparser.add_argument(
+            "--lidheight",  action="store", type=float, default=18,
+            help="height of lid in mm")
+        self.argparser.add_argument(
+            "--hex_hole_corner_radius",  action="store", type=float, default=5,
+            help="The corner radius of the hexagonal dice holes, in mm")
+        self.argparser.add_argument(
+            "--magnet_diameter",  action="store", type=float, default=6,
+            help="The diameter of magnets for holding the box closed, in mm")
+
+    def diceCB(self):
+        t = self.thickness
+        xc = self.y / 2 - t
+        yc = self.x / 2 - t
+        cr = self.hex_hole_corner_radius
+
+        r = (min(self.x, self.y) - 2 * t) / 2 / math.pi
+        apothem = r * math.sqrt(3) / 2
+
+        # dice
```

Comment:
I wonder if this can be done more easier and more generically by using either the new [fillHoles](https://github.com/florianfesti/boxes/blob/master/boxes/__init__.py#L1558) function or one of the old [hex functions](https://github.com/florianfesti/boxes/blob/master/boxes/__init__.py#L1896) probably [hexHolesHex](https://github.com/florianfesti/boxes/blob/master/boxes/__init__.py#L1987).

Reply:
@florianfesti It looks like `hexHolesHex` creates circular holes, in a hexagon pattern, in the shape of a hexagon (rather than filling an entire rectangle with holes). But `fillHoles` could do the job more generically. I don't think that it would be easier, though. I need to get to bed, but I can take a look again tomorrow. Thanks again for all the great suggestions!

Reply:
@florianfesti I experimented with `fillHoles()`, and encountered the following problems:

- the distance between the hexagons horizontally is not the same as the distance between them vertically, so it's not as visually appealing
- there's no way to force the radius of the hexagons to be exactly what we want; there's too much internal manipulation of the values
- it really wants to start by putting a hexagon in the top left corner of the area, thus creating a `3-2-3` arrangement, instead of the `2-3-2` arrangement common to dice boxes
- if you add a new parameter to pass through `corner_radius`, then that parameter doesn't make sense for `round` holes, or for scaled-down holes

For now, it seems better to use `regularPolygonalHole()` directly. This was a good experiment though, because it made me realize that my code also had irregular spacing between the x and y directions. I fixed it by just converting to polar coordinates; it was so much easier, and much clearer what we wanted to accomplish.

Reply:
Yeah, I guess `fillHoles` - which is really new - needs a bit of polishing to be usable here. It currently lacks the means to align the pattern symmetrically with borders and to control which horizontal offset to start with the hex pattern. That on my todo list as it is necessary to replace the older `hex*` functions 

In the long run I can imagine it being used to have an arbitrary number of holes depending on the size of the box. Just hard coding it to 7 holes is fine for now, though.